core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala

```diff
@@ -182,13 +185,13 @@ class ProduceRequestTest extends BaseRequestTest {
 
     // Send the produce request to the non-replica
     val records = MemoryRecords.withRecords(Compression.NONE, new SimpleRecord("key".getBytes, "value".getBytes))
-    val topicPartition = new TopicPartition("topic", partition)
     val produceRequest = ProduceRequest.builder(new ProduceRequestData()
       .setTopicData(new ProduceRequestData.TopicProduceDataCollection(Collections.singletonList(
         new ProduceRequestData.TopicProduceData()
-          .setName(topicPartition.topic())
+          .setName("topic")
```

Comment:
Should we remove `setName`?

Reply:
This is still not removed. It's a bit confusing since there are two ProduceRequestTest, one in scala and another in java. You fixed the one in java, but this one is in scala.